mac/Keyman4MacIM/Keyman4MacIM/KMPackage.m

```diff
@@ -0,0 +1,124 @@
+
+@implementation KMPackage
+
+const NSInteger kUnexpectedFileAsscociationType = 42;
+
++ (BOOL)autosavesInPlace {
+    return NO;
+}
+
+- (void)makeWindowControllers {
+    if ([self.AppDelegate debugMode])
+        NSLog(@"Attempting to display configuration window...");
+    
+    [[self.AppDelegate configWindow] showWindow:nil];
+}
+
+- (KMInputMethodAppDelegate *)AppDelegate {
+    return (KMInputMethodAppDelegate *)[NSApp delegate];
+}
+
+- (NSString *)pathForTemporaryKmpFile:(NSString *)kmpFilename
+{
+    NSString *  result = [NSTemporaryDirectory() stringByAppendingPathComponent:kmpFilename];
+    assert(result != nil);
+    return result;
+}
+
+-(BOOL)readFromFileWrapper:(NSFileWrapper *)fileWrapper ofType:(NSString *)typeName error:(NSError * _Nullable __autoreleasing *)outError {
+    
+    NSString *filename = [fileWrapper filename];
+    
+    if ([self.AppDelegate debugMode])
+        NSLog(@"readFromFileWrapper called with file: %@", filename);
+    
+    if (![typeName isEqualToString: @"Keyman Package"]) {
+        if (outError != NULL) {
+            NSString *description = [@"Unexpected file association for type " stringByAppendingString:typeName];
+            NSDictionary *errorDictionary = @{ NSLocalizedDescriptionKey : description,
+                                               NSFilePathErrorKey : filename
+                                               };
+            *outError = [[NSError alloc] initWithDomain:@"Keyman" code:kUnexpectedFileAsscociationType userInfo:errorDictionary];
+        }
+        return NO;
+    }
+    
+    if (![self userConfirmsInstallationOfPackageFile:filename]) {
+        if ([self.AppDelegate debugMode])
+            NSLog(@"Keyman Package file installation cancelled by user.");
+        return YES; // Returning NO causes the system to report a failure to the user.
+    }
+    
+    NSString *tempFile = [self pathForTemporaryKmpFile:filename];
+    NSURL *fileURL = [[NSURL alloc] initFileURLWithPath:tempFile];
+    if (![fileWrapper writeToURL:fileURL options:NSFileWrapperWritingAtomic originalContentsURL:nil error:outError])
+        return NO;
+    
+    BOOL didUnzip = [self.AppDelegate unzipFile:tempFile];
+    
+    if (!didUnzip && outError != NULL) {
+        NSDictionary *errorDictionary = @{ NSLocalizedDescriptionKey : @"Failed to unzip Keyman Package" };
+        *outError = [[NSError alloc] initWithDomain:@"Keyman" code:kUnexpectedFileAsscociationType userInfo:errorDictionary];
+    }
+    [[NSFileManager defaultManager] removeItemAtPath:tempFile error:nil];
+    return didUnzip;
+}
+
+-(BOOL)userConfirmsInstallationOfPackageFile:(NSString *)fileName {
+    NSAlert *alert = [[NSAlert alloc] init];
+    [alert addButtonWithTitle:NSLocalizedString(@"Yes", @"Alert button")];
+    [alert addButtonWithTitle:NSLocalizedString(@"No", @"Alert button")];
+    [alert setMessageText:NSLocalizedString(@"Install Keyman Package?", @"Alert message text when user double-clicks a KMP file.")];
```

Comment:
I'm not familiar with the dialogs used on Keyman Mac. Do we want some consistency with the Keyman Desktop experience? 

On Windows, the confirmation buttons are "Install" / "Cancel".

![install confirmation](https://user-images.githubusercontent.com/7358010/36237488-9e288bc4-122e-11e8-8be7-215c5b7b5e9e.png)

Reply:
At least currently, I don't even unzip the KMP before asking the question. I could try to make it more like that, but it would be a bit more work. Just changing the button text is easy enough. I'm kind of thinking that if someone double clicks a KMP, they either did it on purpose and know what they're trying to install, or they did it by accident and will click No (or cancel). My inclination is to go ahead and merge what I have working and then consider additional enhancements down the road (possibly for version 10, but given the number of other things that need to be done, probably later).

Reply:
Agree with @tombogle's assessment here. Let's move extra enhancements into a future version.

Reply:
The main reason to have the dialog is to give some continuity when the user clicks it and then it appears in KM Configuration window, correct?